src/dotnet/APIView/APIViewTest/NamedTypeTests.cs

```diff
@@ -0,0 +1,218 @@
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Immutable;
+using System.Linq;
+using APIView;
+using Xunit;
+
+namespace APIViewTest
+{
+    public class NamedTypeTests
+    {
+        [Fact]
+        public void NamedTypeTestClassSomeEventsSomeFieldsNoMethodsSomeNamedTypes()
+        {
+            var reference = MetadataReference.CreateFromFile("TestLibrary.dll");
+            var compilation = CSharpCompilation.Create(null).AddReferences(reference);
+            var a = compilation.SourceModule.ReferencedAssemblySymbols[0];
+
+            var namedTypeSymbol = a.GetTypeByMetadataName("TestLibrary.SomeEventsSomeFieldsNoMethodsSomeNamedTypes");
+            NamedTypeAPIV publicClass = new NamedTypeAPIV(namedTypeSymbol);
+
+            Assert.False(publicClass == null);
+            Assert.Equal("SomeEventsSomeFieldsNoMethodsSomeNamedTypes", publicClass.Name);
+            Assert.Equal("class", publicClass.Type);
+
+            ImmutableArray<EventAPIV> events = publicClass.Events;
+            Assert.Single(events);
+
+            ImmutableArray<FieldAPIV> fields = publicClass.Fields;
+            Assert.Single(fields);
+
+            ImmutableArray<MethodAPIV> methods = publicClass.Methods;
+            Assert.Empty(methods);
+
+            ImmutableArray<NamedTypeAPIV> namedTypes = publicClass.NamedTypes;
+            Assert.Single(namedTypes);
+        }
+
+        [Fact]
+        public void NamedTypeTestClassSomeEventsSomeFieldsNoMethodsSomeNamedTypesStringRep()
+        {
+            var reference = MetadataReference.CreateFromFile("TestLibrary.dll");
+            var compilation = CSharpCompilation.Create(null).AddReferences(reference);
+            var a = compilation.SourceModule.ReferencedAssemblySymbols[0];
+
+            var namedTypeSymbol = a.GetTypeByMetadataName("TestLibrary.SomeEventsSomeFieldsNoMethodsSomeNamedTypes");
+            NamedTypeAPIV publicClass = new NamedTypeAPIV(namedTypeSymbol);
+
+            Assert.Contains("public class SomeEventsSomeFieldsNoMethodsSomeNamedTypes {", publicClass.ToString());
+        }
+
+        [Fact]
+        public void NamedTypeTestInterfaceNoEventsNoFieldsSomeMethodsNoNamedTypes()
+        {
+            var reference = MetadataReference.CreateFromFile("TestLibrary.dll");
+            var compilation = CSharpCompilation.Create(null).AddReferences(reference);
+            var a = compilation.SourceModule.ReferencedAssemblySymbols[0];
+
+            var namedTypeSymbol = a.GetTypeByMetadataName("TestLibrary.PublicInterface`1");
+            NamedTypeAPIV publicInterface = new NamedTypeAPIV(namedTypeSymbol);
+
+            Assert.False(publicInterface == null);
+            Assert.Equal("PublicInterface", publicInterface.Name);
+            Assert.Equal("interface", publicInterface.Type);
+
+            ImmutableArray<EventAPIV> events = publicInterface.Events;
+            Assert.Empty(events);
+
+            ImmutableArray<FieldAPIV> fields = publicInterface.Fields;
```

Comment:
Don't introduce variables where they can be avoided. `Assert.Empty(publicInterface.Fields);` is fine.